### PR TITLE
[IOTDB-2889] Add method converting TsBlock to TSQueryDataSet in QueryDataSetUtils

### PR DESCRIPTION
**Pull request by JackieTien97**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java [hunk 1]
```diff
@@ -204,7 +221,7 @@ public ExecutionResult getStatus() {
     }
   }
 
-  private synchronized void initialResultHandle() throws IOException {
+  private void initialResultHandle() throws IOException {
```
Comment (cornmonster) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java:224]: initial -> init or initialize.

initial means signing (a document) with one's initials in order to authorize or validate it.

File: server/src/main/java/org/apache/iotdb/db/utils/QueryDataSetUtils.java [hunk 2]
```diff
@@ -173,6 +176,180 @@ public static TSQueryDataSet convertQueryDataSetByFetchSize(
     return tsQueryDataSet;
   }
 
+  public static TSQueryDataSet convertTsBlockByFetchSize(
+      QueryExecution queryExecution, int fetchSize) throws IOException {
+    int columnNum = queryExecution.getOutputValueColumnCount();
+    TSQueryDataSet tsQueryDataSet = new TSQueryDataSet();
+    // one time column and each value column has a actual value buffer and a bitmap value to
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/utils/QueryDataSetUtils.java:183]: typo... an actual...

**Review by xingtanzjr — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/utils/QueryDataSetUtils.java (hunk 2 above)
Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/utils/QueryDataSetUtils.java:183]: fix

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java (hunk 1 above)
Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java:224]: fix

**Pull request merged**